google-cloud-storage/src/main/java/com/google/cloud/storage/StorageUtils.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage;
+
+import com.google.cloud.WriteChannel;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Utility methods to perform various operations with the Storage such as upload.
+ *
+ * <p>Example of uploading files from a folder:
+ *
+ * <pre>{@code
+ * File folder = new File("pictures/");
+ * StorageUtils utils = StorageUtils.create(storage);
+ * for (File file: folder.listFiles()) {
+ *     if (!file.isDirectory()) {
+ *         BlobInfo blobInfo = BlobInfo.newBuilder(BUCKET, file.getName()).build();
+ *         try {
+ *             utils.upload(blobInfo, file.toPath());
+ *         } catch (IOException e) {
+ *             System.err.println("Unable to upload " + file + ": " + e.getMessage());
+ *         }
+ *     }
+ * }
+ * }</pre>
+ */
+public final class StorageUtils {
```

Comment:
"Utils" classes usually don't have object state or non-static methods. Perhaps there's a better name here? Uploader or StorageUploader perhaps?

Reply:
This class could be used not only of uploading, it could contain downloading or another functionality as well. Would it be okay to rename it to StorageHelper?

Reply:
That works. Another possibility: StorageTransmitter. 

However, also consider whether the Storage field works better as a method argument.

Reply:
I think it's a very rare case when someone has more than one instance of the storage. So, passing this instance to every method doesn't seem to be convenient. Methods with 4 arguments doesn't look as easy to understand.
Having a class above the storage will certainly give some freedom in extending API without breaking compatibility. To me 'Transmitter' sounds like something closed to the physical layer. 'Helper'  is very generic. May be StorageOperations or StorageFunctions fits better?

Reply:
StorageMover? StorageTransporter?

Reply:
I like StorageOperations, actually.